ext/opentelemetry-ext-redis/src/opentelemetry/instrumentation/redis/__init__.py

```diff
@@ -0,0 +1,61 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""
+Instrument `redis`_ to report Redis queries.
+
+There are two options for instrumenting code. The first option is to use the
+``opentelemetry-auto-instrumentation`` executable which will automatically
+instrument your Redis client. The second is to programmatically enable
+instrumentation via the following code:
+
+.. _redis: https://pypi.org/project/redis/
+
+Usage
+-----
+
+.. code:: python
+
+    from opentelemetry import trace
+    from opentelemetry.instrumentation.redis import RedisInstrumentor
+    from opentelemetry.sdk.trace import TracerProvider
+    import redis
+
+    trace.set_tracer_provider(TracerProvider())
+
+    # Instrument redis
+    RedisInstrumentor().instrument(tracer_provider=trace.get_tracer_provider())
+
+    # This will report a span with the default settings
+    client = redis.StrictRedis(host="localhost", port=6379)
+    client.get("my-key")
+
+API
+---
+"""
+from opentelemetry import trace
+from opentelemetry.auto_instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.instrumentation.redis.patch import _patch, _unpatch
+
+
+class RedisInstrumentor(BaseInstrumentor):
+    """An instrumentor for Redis
+    See `BaseInstrumentor`
+    """
+
+    def _instrument(self, **kwargs):
+        _patch(kwargs.get("tracer_provider", trace.get_tracer_provider()))
```

Comment:
It would be cleaner to get rid of the `_patch` and `_unpatch` methods, since they are just another step that can be avoided here. It also removes another name for the concept of instrumentation, especially since patching is a way of doing instrumentation that applies only to some third party libraries or frameworks.

Reply:
Removed patch/unpatch